miscellaneous/etroc1_cal_studies.py

```diff
@@ -0,0 +1,214 @@
+                fig.add_trace(go.Histogram(
+                    x=df.loc[condition]["calibration_code"],
+                    name='Board {}'.format(board_id), # name used in legend and hover labels
+                    opacity=0.5,
+                    bingroup=1,
+                ))
+            fig.update_layout(
+                barmode='overlay',
+                title_text="Histogram of Even Calibration Code with Odd TOA Code<br><sup>Run: {}{}</sup>".format(run_name, extra_title),
+                xaxis_title_text='Calibration Code', # xaxis label
+                yaxis_title_text='Count', # yaxis label
+            )
+            fig.update_yaxes(type="log")
+
+            fig.write_html(
+                base_path/'calibration_code_histogram_oddTOA_EvenCAL.html',
+                full_html = full_html,
+                include_plotlyjs = 'cdn',
+            )
+
+            # Odd - Odd
+            fig = go.Figure()
+            for board_id in sorted(df["data_board_id"].unique()):
+                condition = (df["data_board_id"] == board_id) & (df["time_of_arrival"]%2 == 1) & (df["calibration_code"]%2 == 1)
+                fig.add_trace(go.Histogram(
+                    x=df.loc[condition]["calibration_code"],
+                    name='Board {}'.format(board_id), # name used in legend and hover labels
+                    opacity=0.5,
+                    bingroup=1,
+                ))
+            fig.update_layout(
+                barmode='overlay',
+                title_text="Histogram of Odd Calibration Code with Odd TOA Code<br><sup>Run: {}{}</sup>".format(run_name, extra_title),
+                xaxis_title_text='Calibration Code', # xaxis label
+                yaxis_title_text='Count', # yaxis label
+            )
+            fig.update_yaxes(type="log")
+
+            fig.write_html(
+                base_path/'calibration_code_histogram_oddTOA_oddCAL.html',
+                full_html = full_html,
+                include_plotlyjs = 'cdn',
+            )    
+
+def script_main(
+    output_directory:Path,
+    make_plots:bool=True,
+    max_toa:float=0,
+    max_tot:float=0,
+    ):
+
+    script_logger = logging.getLogger('apply_event_cuts')
+
+    if max_toa == 0:
+        max_toa = None
+    if max_tot == 0:
+        max_tot = None
+
+    with RM.RunManager(output_directory.resolve()) as Fermat:
+        Fermat.create_run(raise_error=False)
+
+        if not Fermat.task_completed("apply_event_cuts"):
+            raise RuntimeError("You can only run this script after applying event cuts")
+
+        plot_cal_code_task(
+            Fermat,
+            script_logger=script_logger,
+            task_name="plot_calcode_EvenOddTOA_after_cuts",
+            data_file=Fermat.get_task_path("calculate_times_in_ns")/'data.sqlite',
+            filter_files={"event": Fermat.path_directory/"event_filter.fd"},
+        )
```

Comment:
You may want to call this function a second time but with the filter_files parameter empty so that you can get the plots before cuts...